src/components/withSaml/index.jsx

```diff
@@ -0,0 +1,13 @@
+import React, { Component } from 'react';
+import useSiteMetadata from '../../data/useSiteMetadata';
+
+// eslint-disable-next-line react/prefer-stateless-function
+const withSaml = (WrappedComponent) => {
+  const Saml = props => {
+    const { providerSlug } = useSiteMetadata();
```

Comment:
@irfanuddinahmad Did you consider using the `StaticQuery` component from Gatsby instead of this function `useSiteMetadata`? It makes sense to co-locate the query itself within the component. This example snippet would give you the `providerSlug` as a prop to the `IndexPage`:

```jsx
import React from 'react';
import { graphql, StaticQuery } from 'gatsby';

const withSamlQuery = graphql`
  query {
    site {
      siteMetadata {
        providerSlug
      }
    }
  }
`;

const withSaml = WrappedComponent => (
  props => (
    <StaticQuery
      query={withSamlQuery}
      render={data => (
        <WrappedComponent providerSlug={data.site.siteMetadata.providerSlug} {...props} />
      )}
    />
  )
);

export default withSaml;
```

Reply:
Done